datalad/core/distributed/tests/test_push.py

```diff
@@ -261,8 +261,9 @@ def test_push_recursive(
     # running on a clone should make the test scenario more different than
     # test_push(), even for the pieces that should be identical
     top = Clone.__call__(source=origin.path, path=src_path)
-    sub, subsub, subnoannex = top.get(
-        '.', recursive=True, get_data=False, result_xfm='datasets')
+    subs = top.get('.', recursive=True, get_data=False, result_xfm='datasets')
+    # order for '.' should not be relied upon, so sort by path
+    sub, subsub, subnoannex = sorted(subs, key=lambda ds: ds.path)
```

Comment:
Oi. That is a good point. More generally this points to the fact that the order of results is now "random". I think we have a few places in the code that make use of (implicit) order assumptions. I cannot pinpoint any right now, but I am confident that there are a few. Something to be aware of, as those might fail in strange ways, once parallelization is more wide-spread.

Reply:
yeah, I was not 100% certain about this one, i.e. either I should have introduced some additional option `preserver_order=True` to start with -- that would have been most "reliable and conservative".  It would cause a bit more of code complexity and some minor run time (results reporting only I guess, since they would need to "stage" waiting for others to complete) effect. But then thought -- heck, it is against master, so we better spot those assumptions whenever testing breaks, and address them.
But yeah -- if you feel that we better have that option and preserve default current behavior, let me know -- I am ok to implement that one.  Might come handy in some use-cases.

Reply:
note: that option would then need to be channeled to the top interfaces (`get` etc), additional burden on the interface(s) etc... might be easier to spot/fix, insofar it was only few tests (if not one) which had that. For a single path/result - shouldn't matter.  For recursive or on directories invocations  order is anyway cannot be relied upon any ways... so still not sure if more pain than gain in adding it

Reply:
might be worth though adding a test run (or make it a default?) to parallelize by default (to at least 2 or 3).  That could allow to spot other places where order might have been relied upon. (I think I got this one initially whenever I made `get` to parallelize by default, but then I went for conservative `1` as the default, so it would only run producer in a separate thread, but there is only one thread for consumers, so order is preserved now by default)